File: taskhive-backend/src/main/java/com/digiwork/taskhive/module/employee/dto/UpdateEmployeeRequest.java
package com.digiwork.taskhive.module.employee.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateEmployeeRequest {

    private String firstName;

    private String lastName;

    private String phone;

    private String department;

    private String designation;

    private LocalDate joinDate;

    private UUID managerId;
}

File: taskhive_flutter/lib/features/employee/data/models/update_employee_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'update_employee_request.freezed.dart';
part 'update_employee_request.g.dart';

@freezed
class UpdateEmployeeRequest with _$UpdateEmployeeRequest {
  const factory UpdateEmployeeRequest({
    String? firstName,
    String? lastName,
    String? phone,
    String? address,
    String? department,
    String? designation,
    String? joinDate,
  }) = _UpdateEmployeeRequest;

  factory UpdateEmployeeRequest.fromJson(Map<String, dynamic> json) =>
      _$UpdateEmployeeRequestFromJson(json);
}
